feat(discover) Add display modes to discover chart

markstory opened the pull request:
We'd like to expand the number of visualization options we offer. Those new options will be incompatible with previous + release markers. Moving the current options into a dropdown will also enable us to reduce the noise and clutter that release markers and previous period data can introduce.

markstory committed:
feat(discover) Add display modes to discover chart

We'd like to expand the number of visualization options we offer. Those
new options will be incompatible with previous + release markers. Moving
the current options into a dropdown will also enable us to reduce the
noise and clutter that release markers and previous period data can
introduce.

markstory commented on src/sentry/static/sentry/app/views/events/eventsChart.jsx:
```diff
@@ -162,6 +164,52 @@ class EventsChart extends React.Component {
+
+    if (!disableReleases) {
+      const previousChart = chartImplementation;
+      chartImplementation = chartProps => (
```
I don't love this pattern but couldn't come up with something better that wasn't also more complicated/indirect.

markstory commented on src/sentry/static/sentry/app/views/eventsV2/chartFooter.tsx:
```diff
@@ -15,21 +15,27 @@ type Props = {
     total === null ? (
       <SectionValue data-test-id="loading-placeholder" key="total-value">
-        -
+        &mdash;
```
Normalized these loading indicators on an mdash.

markstory committed:
Add a functional test.

wmak commented:
I think this reintroduces some loading jank when you flip between graphs, looks fine still when changing the period though :+1::
![loading jank](https://user-images.githubusercontent.com/4205004/79253977-05ec1f00-7e52-11ea-881c-6bcb4ecc70a1.gif)

markstory commented:
@wmak Yeah, I'm not sure how to get rid of that reload jank just yet.

wmak reviewed (approved):
LGTM

markstory committed:
Merge branch 'master' into feat/discover-display-mode

markstory committed:
Disable animation on charts to remove jank

When animating between states when we replace data causes tearing jank
which is unsightly. Disabling animation removes that jank.

markstory committed:
Fix width of menu so it doesn't change size
